code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/integration/ldap/OpenLDAP/OpenLDAPIdentityProvider.java

```diff
@@ -237,4 +240,22 @@ protected Logger getLogger() {
     public String getName() {
         return getConstantsConfig().getProviderName();
     }
+
+    public String validateIdentities(List<Map<String, String>> identitiesGiven) {
+        List<Identity> identities = getIdentities(identitiesGiven);
+        return openLDAPUtils.toHashSeparatedString(identities);
+    }
+
+    public List<Identity> savedIdentities() {
+        List<String> ids = openLDAPUtils.fromHashSeparatedString(OpenLDAPConstants.LDAP_ALLOWED_IDENTITIES.get());
+        List<Identity> identities = new ArrayList<>();
+        if (ids.isEmpty() || !isConfigured()) {
+            return identities;
+        }
+        for(String id: ids){
+            String[] split = id.split(":", 2);
+            identities.add(getIdentity(split[1], split[0]));
+        }
+        return identities;
+    }
```

Comment:
This method reads the comma separated String of allowedIdentities saved in DB  and fetches each record from OpenLDAP backend and returns a List<Identity>